federation/planner.go

```diff
@@ -145,22 +192,27 @@ func (e *Planner) planObject(typ *graphql.Object, selectionSet *graphql.Selectio
 			return nil, fmt.Errorf("typ %s has no field %s", typ.Name, selection.Name)
 		}
 
-		fieldInfo := e.schema.Fields[field]
+		var hasCustomService bool
+		hasCustomService, localSelections, selectionsByService = e.maybeCustomService(typ.Name, service, selection, localSelections, selectionsByService)
 
-		// Prioritize resolving as many fields as we can in the current service
-		if fieldInfo.Services[service] {
-			localSelections = append(localSelections, selection)
-		} else {
-			serviceWithField := ""
+		if !hasCustomService {
+			fieldInfo := e.schema.Fields[field]
 
-			for service, hasField := range fieldInfo.Services {
-				if hasField {
-					serviceWithField = service
+			// Prioritize resolving as many fields as we can in the current service
+			if fieldInfo.Services[service] {
+				localSelections = append(localSelections, selection)
```

Comment:
I think we're already adding the selection to local selections in `maybeCustomService` if it is in the current service. Should we skip this step?

Reply:
Although I think I might prefer if `maybeCustomService` didn't change the local selections and this was only something we do in this function. That makes it easier to think of `maybeCustomService` as deciding which service to run the field on.

Reply:
If `hasCustomService` is true, we will skip the entire step.